src/Avalonia.Controls/Flyouts/PopupFlyoutBase.cs

```diff
@@ -0,0 +1,518 @@
+
+        /// <inheritdoc cref="Popup.HorizontalOffsetProperty"/>
+        public static readonly StyledProperty<double> HorizontalOffsetProperty =
+            Popup.HorizontalOffsetProperty.AddOwner<PopupFlyoutBase>();
+        
+        /// <inheritdoc cref="Popup.VerticalOffsetProperty"/>
+        public static readonly StyledProperty<double> VerticalOffsetProperty =
+            Popup.VerticalOffsetProperty.AddOwner<PopupFlyoutBase>();
+            
+        /// <inheritdoc cref="Popup.PlacementAnchorProperty"/>
+        public static readonly StyledProperty<PopupAnchor> PlacementAnchorProperty =
+            Popup.PlacementAnchorProperty.AddOwner<PopupFlyoutBase>();
+        
+        /// <inheritdoc cref="Popup.PlacementAnchorProperty"/>
+        public static readonly StyledProperty<PopupGravity> PlacementGravityProperty =
+            Popup.PlacementGravityProperty.AddOwner<PopupFlyoutBase>();
+
+        /// <summary>
+        /// Defines the <see cref="ShowMode"/> property
+        /// </summary>
+        public static readonly DirectProperty<PopupFlyoutBase, FlyoutShowMode> ShowModeProperty =
+            AvaloniaProperty.RegisterDirect<PopupFlyoutBase, FlyoutShowMode>(nameof(ShowMode),
+                x => x.ShowMode, (x, v) => x.ShowMode = v);
+
+        /// <summary>
+        /// Defines the <see cref="OverlayInputPassThroughElement"/> property
+        /// </summary>
+        public static readonly DirectProperty<PopupFlyoutBase, IInputElement?> OverlayInputPassThroughElementProperty =
+            Popup.OverlayInputPassThroughElementProperty.AddOwner<PopupFlyoutBase>(
+                o => o._overlayInputPassThroughElement,
+                (o, v) => o._overlayInputPassThroughElement = v);
+        
+        private readonly Lazy<Popup> _popupLazy;
+        private FlyoutShowMode _showMode = FlyoutShowMode.Standard;
+        private Rect? _enlargedPopupRect;
+        private PixelRect? _enlargePopupRectScreenPixelRect;
+        private IDisposable? _transientDisposable;
+        private Action<IPopupHost?>? _popupHostChangedHandler;
+        private IInputElement? _overlayInputPassThroughElement;
+
+        static PopupFlyoutBase()
+        {
+            Control.ContextFlyoutProperty.Changed.Subscribe(OnContextFlyoutPropertyChanged);
+        }
+
+        public PopupFlyoutBase()
+        {
+            _popupLazy = new Lazy<Popup>(() => CreatePopup());
+        }
+
+        protected Popup Popup => _popupLazy.Value;
+
+        /// <summary>
+        /// Gets or sets the desired placement.
+        /// </summary>
+        public PlacementMode Placement
+        {
+            get => GetValue(PlacementProperty);
+            set => SetValue(PlacementProperty, value);
+        }
+        
+        /// <inheritdoc cref="Popup.PlacementGravity"/>
+        public PopupGravity PlacementGravity
+        {
+            get { return GetValue(PlacementGravityProperty); }
+            set { SetValue(PlacementGravityProperty, value); }
```

Comment:
optional nit: might be good to standardize these to the new syntax (get => GetValue...) for consistency in the file

Reply:
Done